packages/manager/apps/web/client/app/domain-operation/update/domain-operation-update.controller.js

```diff
@@ -192,115 +196,97 @@ angular.module('App').controller(
-                }),
-            );
-          }
-        } else if (!arg.readOnly) {
-          rtn = this.domainOperationService.updateOperation({
-            id: this.operation.id,
-            key: arg.key,
-            data: { value: arg.value },
-          });
-        }
-        return rtn;
-      });
-
-      return this.$q
-        .all(promises)
-        .then(() => {
-          switch (this.todoOperation) {
-            case 'relaunch':
-              return this.domainOperationService[
-                `relaunch${this.constructor.capitalizeFirstLetter(
-                  this.type,
-                )}Operation`
-              ](this.operation.id)
-                .then(() =>
-                  this.Alerter.success(
-                    this.$translate.instant(
-                      'domain_tab_OPERATION_update_relaunch_success',
-                    ),
-                    ALERTER_ID,
-                  ),
-                )
-                .catch((err) =>
-                  this.Alerter.alertFromSWS(
-                    this.$translate.instant('domain_operations_relaunch_error'),
-                    err,
-                    ALERTER_ID,
-                  ),
-                );
-            case 'cancel':
-              return this.domainOperationService[
-                `cancel${this.constructor.capitalizeFirstLetter(
-                  this.type,
-                )}Operation`
-              ](this.operation.id)
-                .then(() =>
-                  this.Alerter.success(
-                    this.$translate.instant('domain_operations_cancel_success'),
-                    ALERTER_ID,
-                  ),
-                )
-                .catch((err) =>
-                  this.Alerter.alertFromSWS(
-                    this.$translate.instant('domain_operations_cancel_error'),
-                    err,
-                    ALERTER_ID,
-                  ),
-                );
-            case 'accelerate':
-              return this.domainOperationService[
-                `accelerate${this.constructor.capitalizeFirstLetter(
-                  this.type,
-                )}Operation`
-              ](this.operation.id)
-                .then(() =>
-                  this.Alerter.success(
-                    this.$translate.instant(
-                      'domain_operations_accelerate_success',
-                    ),
-                    ALERTER_ID,
-                  ),
-                )
-                .catch((err) =>
-                  this.Alerter.alertFromSWS(
-                    this.$translate.instant(
-                      'domain_operations_accelerate_error',
-                    ),
-                    err,
-                    ALERTER_ID,
-                  ),
-                );
-            default:
-              this.Alerter.success(
-                this.$translate.instant('domain_tab_OPERATION_update_success'),
-                ALERTER_ID,
+      for (const arg of this.args) {
```

Comment:
```suggestion
      this.args.forEach ({key, value, type, readOnly}) {
```
I think it's better to use foreach and destructured object

Reply:
with for...of, we can still use destructured objects.

In my opinion, for...of is more readable, and from a performance point of view, it's also better (not a big performance improvement, but an improvement nonetheless).